src/include/txn.i

```diff
@@ -1091,12 +1105,24 @@ __wt_txn_update_check(WT_SESSION_IMPL *session, WT_UPDATE *upd)
         if (upd->txnid != WT_TXN_ABORTED) {
             if (ignore_prepare_set)
                 F_SET(txn, WT_TXN_IGNORE_PREPARE);
-            WT_STAT_CONN_INCR(session, txn_update_conflict);
-            WT_STAT_DATA_INCR(session, txn_update_conflict);
-            return (__wt_txn_rollback_required(session, "conflict between concurrent operations"));
+            return (__txn_rollback_required(session));
         }
     }
 
+    /* If there is no cursor, we don't need to check the on page value. */
+    WT_ASSERT(session, cbt != NULL || (cbt == NULL && !check_onpage_value));
+
+    /* Check conflict against the on page value. */
+    if (check_onpage_value) {
+        WT_RET(__wt_value_return_buf(cbt, cbt->ref, (WT_ITEM *)NULL, &start, &stop));
```

Comment:
I think this `WT_RET` (and the subsequent calls to `__txn_rollback_required`) fail to reset `WT_TXN_IGNORE_PREPARE`, we probably need an `err` label that resets it, something like this:
```
    /*   
     * Always include prepared transactions in this check: they are not supposed to affect
     * visibility for update operations.
     */ 
    ignore_prepare_set = F_ISSET(txn, WT_TXN_IGNORE_PREPARE);
    F_CLR(txn, WT_TXN_IGNORE_PREPARE);
    for (; upd != NULL && !__wt_txn_upd_visible(session, upd); upd = upd->next)  
        if (upd->txnid != WT_TXN_ABORTED)  
            WT_ERR(__txn_rollback_required(session));
        
    /* If there is no cursor, we don't need to check the on page value. */
    WT_ASSERT(session, cbt != NULL || (cbt == NULL && !check_onpage_value));
         
    /* Check conflict against the on page value. */
    if (check_onpage_value) {
        WT_ERR(__wt_value_return_buf(cbt, cbt->ref, (WT_ITEM *)NULL, &start, &stop));
        if (stop.txnid != WT_TXN_MAX && stop.timestamp != WT_TS_MAX &&
          !__wt_txn_visible(session, stop.txnid, stop.timestamp))
            WT_ERR(__txn_rollback_required(session));  
        
        if (!__wt_txn_visible(session, start.txnid, start.timestamp))
            WT_ERR(__txn_rollback_required(session)); 
    }       
            
err:
    if (ignore_prepare_set)
        F_SET(txn, WT_TXN_IGNORE_PREPARE);
    return (ret);
```

Reply:
fixed